asllib/Native.ml

```diff
@@ -254,6 +249,65 @@ module NativeBackend = struct
     let bv = as_bitvector bv in
     Bitvector.length bv |> v_of_int
 
+  let rec unknown_of_aggregate_type unknown_of_singular_type ~eval_expr_sef ty =
+    let unknown_of_type =
+      unknown_of_aggregate_type unknown_of_singular_type ~eval_expr_sef
+    in
+    match ty.desc with
+    | T_Real | T_String | T_Bool | T_Bits _ | T_Int _ ->
+        unknown_of_singular_type ~eval_expr_sef ty
+    | T_Array (length, t) ->
+        let n =
+          match length with
+          | ArrayLength_Enum (_, i) -> i
+          | ArrayLength_Expr e -> (
+              match eval_expr_sef e with
+              | NV_Literal (L_Int n) -> Z.to_int n
+              | _ -> (* Bad types *) assert false)
+        in
+        NV_Vector (List.init n (fun _ -> unknown_of_type t))
+    | T_Record fields | T_Exception fields ->
+        fields
+        |> List.map (fun (field_name, t) -> (field_name, unknown_of_type t))
+        |> IMap.of_list
+        |> fun record -> NV_Record record
+    | T_Enum li ->
+        let n = List.length li |> expr_of_int in
+        let range = Constraint_Range (expr_of_int 0, n) in
+        let t = T_Int (WellConstrained [ range ]) |> add_pos_from ty in
+        unknown_of_singular_type ~eval_expr_sef t
+    | T_Tuple types -> NV_Vector (List.map (fun t -> unknown_of_type t) types)
+    | T_Named _ -> Error.(fatal_from ty TypeInferenceNeeded)
```

Comment:
Why do you fail here? Actually, why don't you first get the structure of the type and then match over the structure?

Reply:
Do you mean that named types are converted to their structure in the typed AST?